src/menu/charprofile.c

```diff
@@ -364,16 +362,17 @@ static bool charprofile_input_handler(SDL_Event *event, void *arg) {
 
 	m->cursor = (m->cursor % m->entries.num_elements) + m->entries.num_elements * (m->cursor < 0);
 
-	if(m->cursor != prev_cursor) {
-		if(ctx->prev_selected_char != m->cursor || dynarray_get(&m->entries, m->cursor).drawdata > 0.95) {
-			ctx->prev_selected_char = prev_cursor;
-			update_char_draw_order(m);
-		}
-	}
 	return false;
 
 }
 
+void preload_charprofile_menu(void) {
+	for(int i = 0; i < NUM_PROFILES-1; i++) {
+		portrait_preload_base_sprite(profiles[i].name, NULL, RESF_PERMANENT);
+		preload_resource(RES_TEXTURE, profiles[i].background, RESF_OPTIONAL);
```

Comment:
Didn't catch this last time, but `RESF_OPTIONAL` is not appropriate here. `RESF_OPTIONAL` makes it so that the game won't crash if this preload fails, but it'll still crash when you actually try to use the failing resource, unless you acquire it through `res_texture_optional()`, e.g.
```c
Texture *tex = res_texture_optional("foobar");

if(tex) {
    use_texture(tex);
} else {
    fallback();
}
```

In 99% cases this is not what you want.